tests/cnf/core/network/metallb/tests/common.go

```diff
@@ -481,8 +505,14 @@ func validatePrefix(
 		}
 	}
 
-	Expect(nextHopAddresses).To(ContainElements(workerNodesAddresses),
-		"Failed next hop address in not in node addresses list")
+	// Check the optional boolean argument
+	if len(noRouteCheck) > 0 && noRouteCheck[0] {
+		Expect(nextHopAddresses).ToNot(ContainElements(workerNodesAddresses),
+			"Failed next hop address should not be in node addresses list")
+	} else {
+		Expect(nextHopAddresses).To(ContainElements(workerNodesAddresses),
+			"Failed next hop address is not in node addresses list")
+	}
```

Comment:
can we combine 2 if-else together after the nexHopAddresses creation?

Reply:
changed